src/SingleArticle.jsx

```diff
@@ -0,0 +1,71 @@
+import { useParams } from "react-router-dom";
+import { getArticleByID } from "./UTILS/utils";
+import { useEffect, useState } from "react";
+import "./CSS/single-article-container.css";
+
+export default function SingleArticle() {
+  const [articleData, setArticleData] = useState({});
+  const { article_id } = useParams();
+
+  useEffect(() => {
+    getArticleByID(article_id)
+      .then((response) => {
+        setArticleData(response.article);
+      })
+      .catch((err) => {
+        console.log(err);
```

Comment:
If we're already catching the error, you could consider adding some error messaging if the request goes wrong. It's also worth considering isLoading state too 👍